Documentation/learning/why.md

```diff
@@ -0,0 +1,14 @@
+# Why etcd
+
+etcd is designed to store a small amount of metadata in consistent, fault-tolerant way. Distributed systems use etcd as the root of scheduling and service discovery configuration storage. [Distributed lock][etcd-concurrency] is implemented on top of etcd, useful for master election in other distributed databases.
+
+Example use cases:
+
+- container linux by CoreOS: Application running on [container linux][container-linux] gets automatic, zero-downtime Linux kernel updates. CoreOS uses [locksmith] to coordinate updates. locksmith uses etcd to implement distributed semaphore, ensuring that only subset of cluster are rebooting at any given time.
```

Comment:
s/container linux/Container Linux
s/CoreOS uses/Container Linux uses

locksmith implements a distributed semaphore over etcd to ensure only a subset of a cluster is rebooting at any given time